### ISSUE TITLE
Feature Request: AWS Budgets

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#753]
_This issue was originally opened by @Stephan1984 as hashicorp/terraform#14313. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hi there,

would be nice, to create AWS Budgets via terraform to monitor costs and resource usage. This is possible via API since Oct 20, 2016: https://aws.amazon.com/de/about-aws/whats-new/2016/10/new-aws-budgets-features-include-api-access-usage-based-budgets-and-more/

API Reference: http://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/operations.html

Kind regards
Stephan

**Comment by netflash** [comment]
I've attached the service list in the same form AWS has them in JSON, e.g.

```
...
    "CostFilters": {
        "Service": [
            "AWS CloudTrail"
        ]
    }
...
```

The advantage of having this list - service names 
- are not documented and
- not consistent across services - some of them have spaces in names, some don't, etc

```
Amazon AppStream
Amazon Athena
Amazon Chime
Amazon Cloud Directory
Amazon CloudFront
Amazon CloudSearch
Amazon Cognito
Amazon Cognito Sync
Amazon Connect
Amazon DynamoDB
Amazon EC2 Container Registry (ECR)
Amazon Elastic Compute Cloud
Amazon Elastic File System
Amazon Elastic MapReduce
Amazon Elastic Transcoder
Amazon ElastiCache
Amazon Elasticsearch Service
Amazon GameLift
Amazon Glacier
Amazon Inspector
Amazon Kinesis
Amazon Kinesis Analytics
Amazon Kinesis Firehose
Amazon Lex
Amazon Lightsail
Amazon Machine Learning
Amazon Mobile Analytics
Amazon Pinpoint
Amazon Polly
Amazon QuickSight
Amazon RDS Service
Amazon Redshift
Amazon Rekognition
Amazon Route 53
Amazon Simple Email Service
Amazon Simple Notification Service
Amazon Simple Queue Service
Amazon Simple Storage Service
Amazon Simple Workflow Service
Amazon SimpleDB
Amazon Virtual Private Cloud
Amazon WorkDocs
Amazon WorkSpaces
Amazon WorkSpaces Application Manager
Amazon Zocalo
AmazonCloudWatch
AmazonWorkMail
AWS Budgets
AWS CloudHSM
AWS CloudTrail
AWS CodeCommit
AWS CodeDeploy
AWS CodePipeline
AWS Config
AWS Data Pipeline
AWS Database Migration Service
AWS Device Farm
AWS Direct Connect
AWS Directory Service
AWS Greengrass
AWS Import/Export
AWS Import/Export Snowball
AWS IoT
AWS Key Management Service
AWS Lambda
AWS OpsWorks
AWS Service Catalog
AWS Shield
AWS Snowball Extra Days
AWS Step Functions
AWS Storage Gateway
AWS WAF
AWS X-Ray
CloudWatch Events
CodeBuild
```

**Comment by StyleT** [comment]
Any news here? :) I would say that this feature is crucial if you have enterprise scale AWS infrastructure.

**Comment by StyleT** [comment]
@xchapter7x Hey! I see that you've started working on this. Looking forward to see it in nearest release! Nice job 😎

**Comment by xchapter7x** [comment]
@StyleT thanks. 
The PR (https://github.com/terraform-providers/terraform-provider-aws/pull/1879) is just waiting on review/feedback from the hashicorp team. 
 Hope it adds value for the community :)

**Comment by bflad** [comment]
The new `aws_bugdets_budget` resource has been merged in via #1879 and will release with v1.17.0 of the AWS provider in about two days.

**Comment by bflad** [comment]
This has been released in [version 1.17.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1170-may-02-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by afeld** [comment]
Issue around Budget notifications: #4548
